blenderproc/python/object/FaceSlicer.py

```diff
@@ -10,6 +10,74 @@
 from blenderproc.python.types.MeshObjectUtility import MeshObject
 from blenderproc.python.utility.Utility import resolve_path
 
+def slice_faces_with_normals(mesh_objects: List[MeshObject], compare_angle_degrees: float = 7.5,
+                  up_vector_upwards: Optional[np.array] = None, new_name_for_object: str = "Surface") \
+        -> List[MeshObject]:
+    """ Extracts normal faces like floors in the following steps:
+    1. Searchs for the specified object.
+    2. Splits the surfaces which point upwards at a specified level away.
+
+    :param mesh_objects: Objects to where all polygons will be extracted.
+    :param compare_angle_degrees: Maximum difference between the up vector and the current polygon normal in degrees.
+    :param up_vector_upwards: If this is True the `up_vec` points upwards -> [0, 0, 1] if not it points downwards: [0, 0, -1] in world coordinates. This vector is used for the `compare_angle_degrees` option.
+    :param new_name_for_object: Name for the newly created object, which faces fulfill the given parameters.
+
+    :return: The extracted surface of the objects.
+    """
+    # set the up_vector
+    if up_vector_upwards is None:
+        up_vector_upwards = np.array([0.0, 0.0, 1.0])
+
+    # the up vector has to have unit length
+    up_vector_upwards /= np.linalg.norm(up_vector_upwards)
+
+    newly_created_objects = []
+    for obj in mesh_objects:
+        obj.edit_mode()
+        bm = obj.mesh_as_bmesh()
+        bpy.ops.mesh.select_all(action='DESELECT')
+
+        list_of_median_poses: List[Tuple[float, bmesh.types.BMFace]] = [(FaceSlicer._get_median_face_pose(f, obj.get_local2world_mat())[2], f)  for f in
+                                bm.faces if FaceSlicer._check_face_angle(f, obj.get_local2world_mat(), up_vector_upwards,
+                                np.deg2rad(compare_angle_degrees))]
+
+        list_of_median_poses_only_z_value = [value for value, face in list_of_median_poses]
+
+        ms = MeanShift(bandwidth=0.005, bin_seeding=True)
```

Comment:
Magic number => give the number a name, this should be the bandwidth in meter, correct?